platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/JsonBundle.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.migratable;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.Map;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.Version;
+
+class JsonBundle {
+  private final String name;
+
+  private final Version version;
+
+  private final long id;
+
+  private final int state;
+
+  private final String location;
+
+  JsonBundle(Map<String, Object> binfo) {
+    this.name = JsonUtils.getStringFrom(binfo, ProfileMigratable.PROFILE_NAME, true);
+    this.id = JsonUtils.getLongFrom(binfo, ProfileMigratable.PROFILE_ID, true);
+    this.state = JsonUtils.getIntegerFrom(binfo, ProfileMigratable.PROFILE_STATE, true);
+    this.location = JsonUtils.getStringFrom(binfo, ProfileMigratable.PROFILE_LOCATION, true);
+    final List<Object> vinfo =
+        JsonUtils.getListFrom(binfo, ProfileMigratable.PROFILE_VERSION, true);
+
+    this.version =
+        new Version( //
+            JsonUtils.getIntegerFrom(vinfo, 0, true),
+            JsonUtils.getIntegerFrom(vinfo, 1, true),
+            JsonUtils.getIntegerFrom(vinfo, 2, true),
+            JsonUtils.getStringFrom(vinfo, 3, true));
+  }
+
+  @VisibleForTesting
+  JsonBundle(String name, Version version, long id, int state, String location) {
+    this.name = name;
+    this.version = version;
+    this.id = id;
+    this.state = state;
+    this.location = location;
+  }
+
+  public static String getName(Bundle bundle) {
+    return bundle.getSymbolicName() + '/' + bundle.getVersion();
+  }
+
+  public static SimpleState getSimpleState(Bundle bundle) {
+    return JsonBundle.getSimpleState(bundle.getState());
+  }
+
+  private static SimpleState getSimpleState(int state) {
```

Comment:
Do we need to account for any of the other Bundle states?

Reply:
Good question, the other states would be INSTALLED and STOPPING which are mapped to INSTALLED. I added a comment to that effect